Automattic/pocket-casts-ios
# Free Trial: Reuse the UpgradeRequiredViewController to display free trial information

Project: #111
Depends on: #112 

## Screenshots
| Before | After Without Trial | After With Trial |
|:---:|:---:|:---:|
|<img width="487" alt="Screen Shot 2022-07-27 at 4 18 33 PM" src="https://user-images.githubusercontent.com/793774/181364483-6f41a26c-3c59-4a5d-a01d-2a54cb42beb5.png">|<img width="487" alt="Screen Shot 2022-07-27 at 4 20 11 PM" src="https://user-images.githubusercontent.com/793774/181364709-46aea0fa-e2d5-4ec0-8d9e-3bde8442db0f.png">|<img width="487" alt="Screen Shot 2022-07-28 at 11 30 17 AM" src="https://user-images.githubusercontent.com/793774/181577619-c1b0c68e-0d15-42f6-b99d-41e1ac633373.png">|





## Description
- Updates the `UpgradeRequiredViewController` to also provide information about the free trial if it's available
- I cleaned up the xib design a bit:
   - Fixed the plus image size since it wasn't set correctly
   - Added a stack view for the title/trial detail label to allow for more consistent spacing between the views
   - Updated the spacing between views to be more consistent
- I took a bold approach in hiding the nav bar to provide a cleaner look to the view 
   - Side effect is the top left X button is no longer there, but the no thanks button does the same function
- I made the title label a bit larger to stand out more
- I added a new method to return "any" free trial duration
   - I didn't want to be stuck requesting a trial for a specific product on the views, this allows us to support trials for any product
   - I wanted to have a priority of display (~yearly then monthly~ monthly then yearly) so I updated the `productIdentifiers` to use the enum values, so I can loop through them in that order. This required some changes to the places where the string values were used. 


## To test

1. Follow the steps from #112 to setup the storekit config, and configure it with a free trial for either product or both to test the any order
2. Launch the app and make sure you don't have plus active
3. Tap on the podcast tab
4. Tap on the new folder button
5. ✅ Verify you see the free trial UI
6. ✅ Verify the title label says the correct duration
7. ✅ Verify the trial subheading label shows with "No payment required - cancel anytime"
8. ✅ Verify the blue button now reads "Start Free Trial"
9. Disable the trials in the storekit config
10. Open the upsell view again
11. ✅ Verify the UI does not reference any trials and the subheading is hidden
12. ✅ Verify the spacing looks good


## Checklist
